BUG: interpolate test_bary_rational.TestAAA.test_basic_functions fails tolerance in chroot with some BLAS

### Describe your issue.

Attempting to build scipy 1.15.1 in an isolated chroot environment (pbuilder), TestAAA.test_basic_functions[<lambda>-7e-13-0] fails tolerance (gets 7.2e-13 instead of < 7e-13)

The test passes on the same system run from the command line, `python3.12 -c "import scipy.interpolate; scipy.interpolate.test(extra_argv=['-k', 'TestAAA', '-v'])"`

So I guess the test is sensitive to runtime conditions, making the tolerance a little too tight.

The problem appears to depend on BLAS.  My local system (which passes) uses OpenBLAS, while the chroot environment which fails uses general netlib BLAS.  I can reproduce the failure locally if I switch my local BLAS implementation to generic netlib BLAS.

Checking other BLAS implementations, it also fails with ATLAS but passes with BLIS.

This must be the basic function test for `lambda x: np.exp(-100*x**2)`.  Looks like it would pass with tolerance 8e-13 (or even 7.3e-13) instead of 7e-13.


### Reproducing Code Example

```python
# using ATLAS or generic netlib BLAS
python3 -c "import scipy.interpolate; scipy.interpolate.test(extra_argv=['-k', 'TestAAA and test_basic_functions', '-v'])"
```

### Error message

```shell
=================================== FAILURES ===================================
________________ TestAAA.test_basic_functions[<lambda>-7e-13-0] ________________

self = <scipy.interpolate.tests.test_bary_rational.TestAAA object at 0x7fae36e1f560>
func = <function TestAAA.<lambda> at 0x7fae36ecf060>, atol = 7e-13, rtol = 0

    @pytest.mark.parametrize("func,atol,rtol",
                             [(lambda x: np.abs(x + 0.5 + 0.01j), 5e-13, 1e-7),
                              (lambda x: np.sin(1/(1.05 - x)), 2e-13, 1e-7),
                              (lambda x: np.exp(-1/(x**2)), 3.5e-13, 0),
                              (lambda x: np.exp(-100*x**2), 7e-13, 0),
                              (lambda x: np.exp(-10/(1.2 - x)), 1e-14, 0),
                              (lambda x: 1/(1+np.exp(100*(x + 0.5))), 2e-13, 1e-7),
                              (lambda x: np.abs(x - 0.95), 1e-6, 1e-7)])
    def test_basic_functions(self, func, atol, rtol):
        with np.errstate(divide="ignore"):
            f = func(PTS)
>       assert_allclose(AAA(UNIT_INTERVAL, func(UNIT_INTERVAL))(PTS),
                        f, atol=atol, rtol=rtol)

atol       = 7e-13
f          = array([3.72007598e-44, 1.53109787e-38, 1.18476388e-33, ...,
       1.18476388e-33, 1.53109787e-38, 3.72007598e-44])
func       = <function TestAAA.<lambda> at 0x7fae36ecf060>
rtol       = 0
self       = <scipy.interpolate.tests.test_bary_rational.TestAAA object at 0x7fae36e1f560>

scipy/interpolate/tests/test_bary_rational.py:211: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

args = (<function assert_allclose.<locals>.compare at 0x7fadf9330c20>, array([ 6.61786764e-14, -5.43790149e-14, -1.03048477e-...), array([3.72007598e-44, 1.53109787e-38, 1.18476388e-33, ...,
       1.18476388e-33, 1.53109787e-38, 3.72007598e-44]))
kwds = {'equal_nan': True, 'err_msg': '', 'header': 'Not equal to tolerance rtol=0, atol=7e-13', 'verbose': True}

    @wraps(func)
    def inner(*args, **kwds):
        with self._recreate_cm():
>           return func(*args, **kwds)
E           AssertionError: 
E           Not equal to tolerance rtol=0, atol=7e-13
E           
E           Mismatched elements: 1 / 1001 (0.0999%)
E           Max absolute difference: 7.27543026e-13
E           Max relative difference: 1.77896034e+30
E            x: array([ 6.617868e-14, -5.437901e-14, -1.030485e-14, ..., -4.372750e-15,
E                  -2.660043e-16,  3.720076e-44])
E            y: array([3.720076e-44, 1.531098e-38, 1.184764e-33, ..., 1.184764e-33,
E                  1.531098e-38, 3.720076e-44])

args       = (<function assert_allclose.<locals>.compare at 0x7fadf9330c20>, array([ 6.61786764e-14, -5.43790149e-14, -1.03048477e-...), array([3.72007598e-44, 1.53109787e-38, 1.18476388e-33, ...,
       1.18476388e-33, 1.53109787e-38, 3.72007598e-44]))
func       = <function assert_array_compare at 0x7fae41e8d4e0>
kwds       = {'equal_nan': True, 'err_msg': '', 'header': 'Not equal to tolerance rtol=0, atol=7e-13', 'verbose': True}
self       = <contextlib._GeneratorContextManager object at 0x7fae41eac1a0>

/usr/lib/python3.12/contextlib.py:81: AssertionError
=========================== short test summary info ============================
FAILED scipy/interpolate/tests/test_bary_rational.py::TestAAA::test_basic_functions[<lambda>-7e-13-0]
= 1 failed, 73828 passed, 4600 skipped, 89 deselected, 325 xfailed, 17 xpassed, 2621 warnings in 1709.56s (0:28:29) =
```

### SciPy/NumPy/Python version and system information

```shell
$ python3.12
Python 3.12.8 (main, Jan 11 2025, 09:42:09) [GCC 14.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import sys, scipy, numpy; print(scipy.__version__, numpy.__version__, sys.version_info); scipy.show_config()
1.15.1 1.26.4 sys.version_info(major=3, minor=12, micro=8, releaselevel='final', serial=0)
Build Dependencies:
  blas:
    detection method: pkgconfig
    found: true
    include directory: /usr/include/x86_64-linux-gnu
... openblas appears here from the local build time configuration
... the runtime error occurs when using /usr/lib/x86_64-linux-gnu/atlas/
    lib directory: /usr/lib/x86_64-linux-gnu/openblas-pthread
    name: blas
    openblas configuration: unknown
    pc file directory: /usr/lib/x86_64-linux-gnu/pkgconfig
    version: 0.3.28+ds
  lapack:
    detection method: pkgconfig
    found: true
    include directory: /usr/include/x86_64-linux-gnu
    lib directory: /usr/lib/x86_64-linux-gnu/openblas-pthread
    name: lapack
    openblas configuration: unknown
    pc file directory: /usr/lib/x86_64-linux-gnu/pkgconfig
    version: 0.3.28+ds
  pybind11:
    detection method: pkgconfig
    include directory: /include
    name: pybind11
    version: 2.13.6
Compilers:
  c:
    args: -g, -O2, -Werror=implicit-function-declaration, -ffile-prefix-map=/projects/python/build/scipy=.,
      -fstack-protector-strong, -fstack-clash-protection, -Wformat, -Werror=format-security,
      -fcf-protection, -Wdate-time, -D_FORTIFY_SOURCE=2
    commands: cc
    linker: ld.bfd
    linker args: -Wl,-z,relro, -g, -O2, -Werror=implicit-function-declaration, -ffile-prefix-map=/projects/python/build/scipy=.,
      -fstack-protector-strong, -fstack-clash-protection, -Wformat, -Werror=format-security,
      -fcf-protection, -Wdate-time, -D_FORTIFY_SOURCE=2
    name: gcc
    version: 14.2.0
  c++:
    args: -g, -O2, -ffile-prefix-map=/projects/python/build/scipy=., -fstack-protector-strong,
      -fstack-clash-protection, -Wformat, -Werror=format-security, -fcf-protection,
      -Wdate-time, -D_FORTIFY_SOURCE=2
    commands: c++
    linker: ld.bfd
    linker args: -Wl,-z,relro, -g, -O2, -ffile-prefix-map=/projects/python/build/scipy=.,
      -fstack-protector-strong, -fstack-clash-protection, -Wformat, -Werror=format-security,
      -fcf-protection, -Wdate-time, -D_FORTIFY_SOURCE=2
    name: gcc
    version: 14.2.0
  cython:
    commands: cython
    linker: cython
    name: cython
    version: 3.0.11
  fortran:
    args: -fPIC
    commands: gfortran
    linker: ld.bfd
    linker args: -Wl,-z,relro, -fPIC
    name: gcc
    version: 14.2.0
  pythran:
    include directory: ../../../../../usr/lib/python3/dist-packages/pythran
    version: 0.17.0
Machine Information:
  build:
    cpu: x86_64
    endian: little
    family: x86_64
    system: linux
  cross-compiled: false
  host:
    cpu: x86_64
    endian: little
    family: x86_64
    system: linux
Python Information:
  path: /usr/bin/python3.12
  version: '3.12'
```

--- a/scipy/interpolate/tests/test_bary_rational.py
+++ b/scipy/interpolate/tests/test_bary_rational.py
@@ -201,7 +201,7 @@ class TestAAA:
                              [(lambda x: np.abs(x + 0.5 + 0.01j), 5e-13, 1e-7),
                               (lambda x: np.sin(1/(1.05 - x)), 2e-13, 1e-7),
                               (lambda x: np.exp(-1/(x**2)), 3.5e-13, 0),
-                              (lambda x: np.exp(-100*x**2), 7e-13, 0),
+                              (lambda x: np.exp(-100*x**2), 8e-13, 0),
                               (lambda x: np.exp(-10/(1.2 - x)), 1e-14, 0),
                               (lambda x: 1/(1+np.exp(100*(x + 0.5))), 2e-13, 1e-7),
                               (lambda x: np.abs(x - 0.95), 1e-6, 1e-7)])

